packages/next/build/swc/src/styled_jsx/utils.rs

```diff
@@ -0,0 +1,368 @@
+  let opening = JSXOpeningElement {
+    name: JSXElementName::Ident(Ident {
+      sym: style_import_name.to_string().into(),
+      span: DUMMY_SP,
+      optional: false,
+    }),
+    attrs,
+    span: DUMMY_SP,
+    self_closing: false,
+    type_args: None,
+  };
+
+  let closing = Some(JSXClosingElement {
+    name: JSXElementName::Ident(Ident {
+      sym: style_import_name.to_string().into(),
+      span: DUMMY_SP,
+      optional: false,
+    }),
+    span: DUMMY_SP,
+  });
+
+  let children = vec![JSXElementChild::JSXExprContainer(JSXExprContainer {
+    expr: JSXExpr::Expr(Box::new(Expr::Ident(style.identifier.clone()))),
+    span: DUMMY_SP,
+  })];
+  JSXElement {
+    opening,
+    closing,
+    children,
+    span: DUMMY_SP,
+  }
+}
+
+pub fn make_local_styled_jsx_el(
+  style_info: &LocalStyle,
+  css_expr: Expr,
+  style_import_name: &String,
+) -> JSXElement {
+  let mut attrs = vec![JSXAttrOrSpread::JSXAttr(JSXAttr {
+    name: JSXAttrName::Ident(Ident {
+      sym: "id".into(),
+      span: DUMMY_SP,
+      optional: false,
+    }),
+    value: Some(JSXAttrValue::JSXExprContainer(JSXExprContainer {
+      expr: JSXExpr::Expr(Box::new(string_literal_expr(
+        hash_string(&style_info.hash).clone().as_str(),
+      ))),
+      span: DUMMY_SP,
+    })),
+    span: DUMMY_SP,
+  })];
+
+  if style_info.is_dynamic {
+    attrs.push(JSXAttrOrSpread::JSXAttr(JSXAttr {
+      name: JSXAttrName::Ident(Ident {
+        sym: "dynamic".into(),
+        span: DUMMY_SP,
+        optional: false,
+      }),
+      value: Some(JSXAttrValue::JSXExprContainer(JSXExprContainer {
+        expr: JSXExpr::Expr(Box::new(Expr::Array(ArrayLit {
+          elems: style_info
+            .expressions
+            .iter()
+            .map(|expression| {
+              Some(ExprOrSpread {
+                expr: expression.clone(),
+                spread: None,
+              })
+            })
+            .collect(),
+          span: DUMMY_SP,
+        }))),
+        span: DUMMY_SP,
+      })),
+      span: DUMMY_SP,
+    }));
+  }
+
+  let opening = JSXOpeningElement {
+    name: JSXElementName::Ident(Ident {
+      sym: style_import_name.to_string().into(),
+      span: DUMMY_SP,
+      optional: false,
+    }),
+    attrs,
+    span: DUMMY_SP,
+    self_closing: false,
+    type_args: None,
+  };
+
+  let closing = Some(JSXClosingElement {
+    name: JSXElementName::Ident(Ident {
+      sym: style_import_name.to_string().into(),
+      span: DUMMY_SP,
+      optional: false,
+    }),
+    span: DUMMY_SP,
+  });
+
+  let children = vec![JSXElementChild::JSXExprContainer(JSXExprContainer {
+    expr: JSXExpr::Expr(Box::new(css_expr)),
+    span: DUMMY_SP,
+  })];
+  JSXElement {
+    opening,
+    closing,
+    children,
+    span: DUMMY_SP,
+  }
+}
+
+pub fn get_usable_import_specifier(_items: &Vec<ModuleItem>) -> String {
+  // TODO
+  String::from("_JSXStyle")
+}
+
+pub fn styled_jsx_import_decl(style_import_name: &String) -> ModuleItem {
```

Comment:
You don't have to accept `&String`.

```rust

fn foo(s: &str) {
}

fn main() {
    let s = String::new();
    foo(&s);
}
```

just works, because `String` can be dereferenced to `&str`. (i.e. `String` implements `Deref<Target = str>`)